powerauth-push-server/src/main/java/io/getlime/push/service/InboxService.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2022 Wultra s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getlime.push.service;
+
+import io.getlime.push.errorhandling.exceptions.InboxMessageNotFoundException;
+import io.getlime.push.model.entity.ListOfInboxMessages;
+import io.getlime.push.model.request.CreateInboxMessageRequest;
+import io.getlime.push.model.response.GetInboxMessageCountResponse;
+import io.getlime.push.model.response.GetInboxMessageDetailResponse;
+import io.getlime.push.repository.InboxRepository;
+import io.getlime.push.repository.converter.InboxMessageConverter;
+import io.getlime.push.repository.model.InboxMessageEntity;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * Service class for tasks related to the inbox.
+ *
+ * @author Petr Dvorak, petr@wultra.com
+ */
+@Service
+public class InboxService {
+
+    private final InboxRepository inboxRepository;
+    private final InboxMessageConverter inboxMessageConverter;
+
+    @Autowired
+    public InboxService(InboxRepository inboxRepository, InboxMessageConverter inboxMessageConverter) {
+        this.inboxRepository = inboxRepository;
+        this.inboxMessageConverter = inboxMessageConverter;
+    }
+
+    @Transactional
+    public GetInboxMessageDetailResponse postMessage(String userId, CreateInboxMessageRequest request) {
+        final InboxMessageEntity messageEntity = inboxMessageConverter.convert(UUID.randomUUID(), userId, request, new Date());
+        final InboxMessageEntity savedMessageEntity = inboxRepository.save(messageEntity);
+        return inboxMessageConverter.convertResponse(savedMessageEntity);
+    }
+
+    public ListOfInboxMessages fetchMessageListForUser(String userId, boolean onlyUnread, Pageable pageable) {
```

Comment:
I would add `@Transactional(readOnly=true)` or a comment why transactions are not required for reading.

Reply:
I added the annotation.